animations/src/main/java/com/chiralbehaviors/CoRE/phantasm/impl/PhantasmDefinition.java

```diff
@@ -94,31 +92,31 @@ public PhantasmDefinition(Class<Phantasm<RuleForm>> phantasm) {
 
     @SuppressWarnings("unchecked")
     public Phantasm<?> wrap(ExistentialRuleform<?, ?> ruleform, Model model) {
-        Map<Class<?>, Object> stateMap = new HashMap<>();
-        Object[] instances = new Object[facets.size()];
-        int i = 0;
-        stateMap.put(Phantasm.class, ruleform);
-        Object thisState = null;
-        for (StateDefinition<RuleForm> facet : facets) {
-            Object state = facet.wrap((RuleForm) ruleform, model);
-            if (facet.equals(thisFacet)) {
-                thisState = state;
-            }
-            instances[i++] = state;
-            stateMap.put(facet.getStateInterface(), state);
+        for (StateDefinition<RuleForm> facet : facets.values()) {
+            facet.constrain(model, (RuleForm) ruleform);
         }
-        if (ScopedPhantasm.class.isAssignableFrom(phantasm)) {
-            if (thisFacet == null) {
-                throw new IllegalStateException(
-                                                String.format("%s extends ScopedPhantasm, but does not define @State annotation",
-                                                              phantasm));
+        PhantasmTwo<RuleForm> doppelgänge = new PhantasmTwo<RuleForm>(
```

Comment:
I support your aggressive tactics to encourage the use of autocomplete, but it's spelled doppelgänger. With an 'r'.